File: main-project/Models/Booking.cs
namespace main_project.Models;

public class Booking
{
    public int Id { get; set; }
    public int RenterId { get; set; }
    public int HostelId { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public string Status { get; set; } = "pending";
    public int? LeaseContractId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Hostel? Hostel { get; set; }
    public RenterProfile? RenterProfile { get; set; }

    public static IQueryable<Booking> Query()
    {
        DatabaseContext db = new DatabaseContext();
        return db.bookings;
    }
}

File: crawler/src/models/booking.ts
import { DataTypes } from "sequelize";
import sequelize from "../database/database";
import RenterProfile from "./renter-profile";
import LeaseContract from "./lease-contract";
import Hostel from "./hostel";

const Booking = sequelize.define("Booking", {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    renterId: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: RenterProfile,
            key: "id",
        },
    },
    hostelId: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: Hostel,
            key: "id",
        },
    },
    startDate: {
        type: DataTypes.DATE,
        allowNull: false,
    },
    endDate: {
        type: DataTypes.DATE,
        allowNull: false,
    },
    status: {
        type: DataTypes.ENUM("pending", "approved", "rejected", "cancelled"),
        allowNull: false,
        defaultValue: "pending",
    },
    leaseContractId: {
        type: DataTypes.INTEGER,
        allowNull: true,
        references: {
            model: LeaseContract,
            key: "id",
        },
    },
}, {
    timestamps: true,
    tableName: "bookings",
    underscored: true,
});

export default Booking;